cpp/src/arrow/engine/substrait/options.h

```diff
@@ -93,6 +106,11 @@ struct ARROW_ENGINE_EXPORT ConversionOptions {
   /// The default behavior will return an invalid status if the plan has any
   /// named table relations.
   NamedTableProvider named_table_provider = kDefaultNamedTableProvider;
+  /// \brief A custom strategy to be used for mapping a tap kind to a function name
+  ///
+  /// The default behavior will return an invalid status if the plan has any
+  /// named tap relations.
```

Comment:
Couldn't a default implementation pass the `kind` through directly?  E.g. `pass_for_named_tap -> pass_for_named_tap`?